ProjectSidewalk/SidewalkWebpage
# 1395 quality control validation interface on mobile

Resolves #1395.
**Changes**
![image](https://user-images.githubusercontent.com/16949591/54585881-97335c80-49d8-11e9-8a86-06986b7302a8.png)
![image](https://user-images.githubusercontent.com/16949591/54585854-8c78c780-49d8-11e9-88b6-25ed6650586c.png)
![image](https://user-images.githubusercontent.com/16949591/54585902-a1edf180-49d8-11e9-8fe7-0dd692b895d7.png)
![image](https://user-images.githubusercontent.com/16949591/58101702-f3f7f280-7b94-11e9-8e28-1fd6d7119901.png)
![image](https://user-images.githubusercontent.com/16949591/58101829-230e6400-7b95-11e9-9134-d00dbc9a0ebd.png)
![image](https://user-images.githubusercontent.com/16949591/58101918-476a4080-7b95-11e9-8913-9f14d206a0a6.png)

- created mobile validation interface that allows users of mobile to complete validation missions
- allows users to see more information for each label by clicking yellow 'i' button
- allows users to leave feedback from the information screen
- allows users to skip to next label
- allows users to interact with GSV using touch screen
- includes basic UI like a mission welcome screen, a mission complete screen, a progress bar, etc.
- will populate every time a user visits any Sidewalk webpage on mobile

**TODOs before merging/known bugs**
- [x]  Adapt PanoMarker algorithm to account for mobile screen size
- [x] Either hide or implement back button functionality
- [x]  Log that validation comes from mobile vs web
- [x]  Prevent landscape mode
- [x] Style "Yes," "No," "Not Sure" buttons

**Testing**
To test, all you need to do is connect to localhost via your phone:
1. Find your computer's IP address
2. Go to your browser with IP_ADDRESS:9000/
3. Start playing!

Things to test:
- mission modal pops up as expected every time the page loads and describes the right mission
- progress bar works as expected
- skip button works as expected
- info button works as expected and brings up the right instructions for the current mission
- leaving a comment works and is reflected in the DB
- validating a label works and is reflected in the DB
- the end of missions screen shows up correctly, with the right number of "agree" "disagree" and "Not sure" labels
- the next mission loads correctly
- cannot validate labels in landscape mode
- interactions are correctly logged, and DB stores if an action occurred in mobile or web
- the browser and mobile implementation rely on each other. make sure that the changes for mobile do not interfere with the browser version and vice versa.
